lib/broccoli/ember-app.js

```diff
@@ -682,34 +682,46 @@ class EmberApp {
     return this._addonTreesFor(type).map(addonBundle => addonBundle.tree);
   }
 
-  _getDefaultCompiler(type) {
-    // since we are now using the app's registry to compile addon code,
-    // we could get into the case where the app has multiple compilers (coffeescript, emblem, etc.)
-    // operating on addon code where there weren't previously.
-    // This prevents that by using the default app compiler only.
+  _getDefaultPluginForType(type) {
     let plugins = this.registry.load(type);
-    return plugins[plugins.length - 1];
+    let defaultsForType = plugins.filter(plugin => plugin.isDefaultForType);
+
+    if (defaultsForType.length > 1) {
+      throw new Error(
+        `There are multiple preprocessor plugins that are registered for '${type}': ${defaultsForType.map(p => p.name).join(', ')}`
```

Comment:
We should probably mention `... marked as default ...`, otherwise it doesn't seem very actionable.